internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -372,7 +374,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// after logging, publishing an event and updating the SYNC status condition
 	if meta.IsPaused(cm) {
 		r.record.Event(cm, event.Normal(reasonPaused, "Reconciliation is paused via the pause annotation"))
-		cm.SetConditions(xpv1.ReconcilePaused())
+		cm.SetConditions(v1.ReconcilePaused())
+		if meta.WasDeleted(cm) {
+			log.Info("Deletion requested, but blocked by the pause annotation, remove it to unblock the request", "annotation", meta.AnnotationKeyReconciliationPaused, "value", "true")
+			r.record.Event(cm, event.Normal(reasonPaused, "Deletion requested, but blocked by the pause annotation, remove it to unblock the request"))
+		}
```

Comment:
```suggestion
		r.record.Event(cm, event.Normal(reasonPaused, "Reconciliation (including deletion) is paused via the pause annotation"))
		cm.SetConditions(v1.ReconcilePaused())
```

It should be enough to update the message as above, no need for special handling for `meta.wasDeleted`.

After this message, it is clear that:

- Reconciliation is paused.
- This includes deletion.
- It is via pause annotation, so, you can remove it to resume. It is the same for both resuming deleted and non-deleted resources.

Reply:
Updated the PR with the suggested changes. I assumed I should apply it to all places in this PR. Please let me know if my understanding was correct.

> It should be enough to update the message as above, no need for special handling for meta.wasDeleted.

It should be enough, but this degrades the previously attempted user experience improvements. Namely:
* the logs are generated on DEBUG level, making `Reconciliation (including deletion) is paused via the pause annotation"` message invisible by default. The previous attempt wanted to bring them to INFO level, so that users can quickly spot them in the logs
* the current message is rather vague, refers to `pause annotation`. IMHO, we should be as much as explicit to users, so that they do not need to checks docs to understand what pause annotation is. Writing the exact annotation in the message is super helpful when one person pauses reconciliation, but another one performs some cluster maintenance/cleanup. In large orgs that is a common case. That person does not even need to know what Crossplane is, but the right message would unblock them quickly.
* Having an extra message that deletion is attempted on paused object would further help by debugging.

Reply:
Thanks!

> - the logs are generated on DEBUG level, making Reconciliation (including deletion) is paused via the pause annotation" message invisible by default. The previous attempt wanted to bring them to INFO level, so that users can quickly spot them in the logs

I agree that INFO would be better here 👍 

> - the current message is rather vague, refers to pause annotation. IMHO, we should be as much as explicit to users, so that they do not need to checks docs to understand what pause annotation is. Writing the exact annotation in the message is super helpful when one person pauses reconciliation, but another one performs some cluster maintenance/cleanup. In large orgs that is a common case. That person does not even need to know what Crossplane is, but the right message would unblock them quickly.

Users can just check the annotations on the resource and figure out which is the pause annotation without necessarily checking the documentation. However, being more explicit wouldn't hurt.

> - Having an extra message that deletion is attempted on paused object would further help by debugging.

Just like the above, I would agree with this in the context of being more explicit. However, I don't think this is worth introducing an additional branch (i.e. `if`) to the logic.